Assets/MRTK/Providers/WindowsMixedReality/XR2018/Controllers/HPMotionController.cs

```diff
@@ -0,0 +1,309 @@
+            new MixedRealityInteractionMapping(1, "Grip Press", AxisType.SingleAxis, DeviceInputType.TriggerPress),
+            new MixedRealityInteractionMapping(2, "Trigger Press", AxisType.Digital, DeviceInputType.Select),
+            new MixedRealityInteractionMapping(3, "Button.X Press", AxisType.Digital, DeviceInputType.PrimaryButtonPress),
+            new MixedRealityInteractionMapping(4, "Button.Y Press", AxisType.Digital, DeviceInputType.SecondaryButtonPress),
+            new MixedRealityInteractionMapping(5, "Menu Press", AxisType.Digital, DeviceInputType.Menu),
+            new MixedRealityInteractionMapping(6, "Thumbstick Position", AxisType.DualAxis, DeviceInputType.ThumbStick),
+            new MixedRealityInteractionMapping(7, "Thumbstick Press", AxisType.Digital, DeviceInputType.ThumbStickPress),
+        };
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultRightHandedInteractions => new[]
+        {
+            new MixedRealityInteractionMapping(0, "Spatial Pointer", AxisType.SixDof, DeviceInputType.SpatialPointer),
+            new MixedRealityInteractionMapping(1, "Grip Press", AxisType.SingleAxis, DeviceInputType.TriggerPress),
+            new MixedRealityInteractionMapping(2, "Trigger Press", AxisType.Digital, DeviceInputType.Select),
+            new MixedRealityInteractionMapping(3, "Button.A Press", AxisType.Digital, DeviceInputType.PrimaryButtonPress),
+            new MixedRealityInteractionMapping(4, "Button.B Press", AxisType.Digital, DeviceInputType.SecondaryButtonPress),
+            new MixedRealityInteractionMapping(5, "Menu Press", AxisType.Digital, DeviceInputType.Menu),
+            new MixedRealityInteractionMapping(6, "Thumbstick Position", AxisType.DualAxis, DeviceInputType.ThumbStick),
+            new MixedRealityInteractionMapping(7, "Thumbstick Press", AxisType.Digital, DeviceInputType.ThumbStickPress),
+        };
+
+        private static readonly ProfilerMarker UpdateControllerPerfMarker = new ProfilerMarker("[MRTK] HPController.UpdateController");
+
+        public override void UpdateController(InteractionSourceState interactionSourceState)
+        {
+            if (MotionControllerState != null)
+            {
+                // If the Motion controller state is instantiated and tracked, use it to update the interaction bool data and the interaction source to update the 6-dof data
+                UpdateController(MotionControllerState);
+                UpdateSixDofData(interactionSourceState);
+            }
+            else
+            {
+                // Otherwise, update normally
+                base.UpdateController(interactionSourceState);
+            }
+        }
+
+        /// <summary>
+        /// Update the controller data from .
+        /// </summary>
+        internal virtual void UpdateController(MotionControllerState controllerState)
+        {
+            using (UpdateControllerPerfMarker.Auto())
```

Comment:
Looks like this method has this line twice (here and line 99). It might be better to be moved to `UpdateController(InteractionSourceState interactionSourceState)` so it wraps both this method and the base call, regardless of which is called)